src/Type/TypeCombinator.php

```diff
@@ -708,6 +713,10 @@ private static function processArrayTypes(array $arrayTypes): array
 		}
 
 		if ($generalArrayOccurred && (!$overflowed || $filledArrays > 1)) {
+			$reducedArrayTypes = self::reduceArrays($arrayTypes, false);
+			if (count($reducedArrayTypes) === 1) {
+				return [self::intersect($reducedArrayTypes[0], ...$accessoryTypes)];
```

Comment:
As far as I can tell, currently the accessory types will always be empty (so I can't test this). But I added them anyway to be safe in case new accessory types are added in the future.